diff --git a/sipbuild/generator/parser/rules.py b/sipbuild/generator/parser/rules.py
index c221ad7..e79e1b8 100644
--- a/sipbuild/generator/parser/rules.py
+++ b/sipbuild/generator/parser/rules.py
@@ -2293,6 +2293,7 @@ def p_opt_base_exception(p):
 
     if len(p) == 4:
         base = p[2]
+        base.make_absolute()
 
         # See if it is a project-defined exception.
         for xd in p.parser.pm.spec.exceptions:
